text/0000-deprecations.md

```diff
@@ -0,0 +1,105 @@
+- Feature Name: deprecations
+- Start Date: 2016-02-15
+- RFC PR: [https://github.com/gruntjs/grunt/pull/1625](https://github.com/gruntjs/grunt/pull/1625)
+- Grunt Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Create a private deprecation API and mark all the APIs in Grunt 1.x.x that are
+deprecated.
+
+# Motivation
+[motivation]: #motivation
+
+Many APIs have been deprecated but it's not entirely clear nor communicated to
+the end users. During 1.x.x, we can add a deprecation API that hooks into any
+deprecated API and logs a message informing the end user.
+
+# Detailed design
+[design]: #detailed-design
+
+A tool to mark an API as deprecated could be:
+
+```js
+var deprecate = require('./lib/grunt/deprecate.js');
+deprecate(grunt.util, '_', 'grunt.util._ has been deprecated. Please use "lodash" directly: https://gruntjs.com/migration-guide#lodash');
+```
+
+Then if any plugin or Gruntfile uses that API, it will log the message:
+
+```shell
+>> grunt.util._ has been deprecated. Please use lodash directly: https://gruntjs.com/migration-guide#lodash
+```
+
+If `--stack` is provide, it will display a stack trace to located where the
+deprecated API.
+
+### Disable Deprecation Notices
+
+All deprecation notifications can be disabled with a cli flag:
+
+```shell
+grunt foo --hide-deprecations
```

Comment:
how about an ENV variable instead of `--hide-deprecations`? or both

Reply:
I'd vote no because AFAIK we haven't done that with any other Grunt options in the past. People can do `grunt.option('hide-deprecations', process.env.GRUNT_HIDE_DEPRECATIONS)` in their Gruntfile if needed too.